### PR TITLE
Raise exceptions for disallowed auto-ivc configurations..

### LINKED ISSUE #1516: Add Error message for two cases that need manual IVC.
### Summary of Issue

Add Error message for two cases that need manual IVC.

### Issue Type

- [x ] Bug
- [ ] Enhancement  
      POEM #___ <N/A or POEM number at https://github.com/OpenMDAO/POEMs>

### Description

1. Manual IVCs are required for distributed component inputs.
2. Manual IVCs are required for inputs promoted together if src_indices are defined on all of them.

### Environment

<For bugs, otherwise N/A>

Operating System: <i.e. OS X 10.14.6, Windows 10, Ubuntu 16.04>

Python environment: <i.e. Anaconda Python 3.7.1>

Packages: <versions of Dymos, OpenMDAO, and other pertinent Python packages. You can just paste the output of `pip freeze` or `conda list` here.>

### PR DESCRIPTION
**Pull request by Kenneth-T-Moore**
### Summary

Raise an exception if when OpenMDAO tries to create an automatic IndepVarComp ouput for these cases:

1. Any input on a distributed component.
2. Any inputs promoted together with src_indices if there isn't at least one that is full width.

Also corrected a couple of sizes in scipyoptimizer.

### Related Issues

- Resolves #1516

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

File: openmdao/docs/features/core_features/defining_components/distributed_comps.rst [hunk 1]
```diff
@@ -71,6 +71,20 @@ This example is run with two processes and a :code:`size` of 15:
     openmdao.core.tests.test_distribcomp.MPIFeatureTests.test_distribcomp_feature
     :layout: interleave
 
+In this example, we introduce a new component called an ``IndepVarComp``. If you used OpenMDAO prior to
+version 3.2, then you have are familiar with this component. An IndepVarComp is used to define an
+independent variable. You no longer have to define these because OpenMDAO defines and uses them
+automatically for all inputs in your model. However, when we define a distributed input, we often
```
Comment (naylor-b) [inline openmdao/docs/features/core_features/defining_components/distributed_comps.rst:77]: all inputs --> all unconnected inputs

File: openmdao/docs/features/core_features/defining_components/distributed_comps.rst [hunk 2]
```diff
@@ -71,6 +71,20 @@ This example is run with two processes and a :code:`size` of 15:
...
+use the "src_indices" attribute to determine the allocation of that input to the processors that the
+component sees. For some sets of these indices, it isn't possible to easilly determine the full size
```
Comment (naylor-b) [inline openmdao/docs/features/core_features/defining_components/distributed_comps.rst:79]: typo: easilly

Comment (Kenneth-T-Moore) [reply openmdao/docs/features/core_features/defining_components/distributed_comps.rst]: All fixed.

File: openmdao/core/group.py [hunk 3]
```diff
@@ -2598,102 +2597,31 @@ def _get_auto_ivc_out_val(self, tgts, remote_vars, all_abs2meta, abs2meta):
...
-                        dist_ranges.append((tgt, self.comm.allgather((0, 0, 0))))
-                else:  # assume default distrib layout
-                    if tgt in abs2meta:
-                        info.append((tgt, abs2meta[tgt]['size'], abs2meta[tgt]['value'], True))
-                    else:
-                        info.append((tgt, 0, np.zeros(0), True))
+                # OpenMDAO currently can't create an automatic IndepVarComp for inputs on
+                # distributed components.
+                msg = 'Distributed component input "{}" cannot be connected to an automatic '
+                msg += 'IndepVarComp.  Please add one manually and connect it.'
```
Comment (swryan) [inline openmdao/core/group.py:2610]: I think that the whole "AutoIVC thing is supposed to be transparent to the user?
May be better to just say that distributed components require an IndepVarComp?

File: openmdao/core/group.py [hunk 4]
```diff
@@ -2598,102 +2597,31 @@ def _get_auto_ivc_out_val(self, tgts, remote_vars, all_abs2meta, abs2meta):
...
-                        break
-                    if sz > 0:
-                        nz_count += 1
-
-            info.append((tgt, val.size, val, True))
+        if src_idx_found:  # auto_ivc connected to local vars with src_indices
+            tgts = ', '.join(src_idx_found)
+            msg = 'The following inputs [{}] are defined using src_indices. These cannot be '
+            msg += 'connected to an automatic IndepVarComp because the source size is '
+            msg += 'undetermined.  Please add one manually and connect it.'
```
Comment (swryan) [inline openmdao/core/group.py:2623]: I think that the whole "AutoIVC thing is supposed to be transparent to the user?
May be better to just say that an IndepVarComp is required to determine the source size?

Comment (Kenneth-T-Moore) [reply openmdao/core/group.py]: Simplified both messages.

**Pull request merged**